coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -1831,7 +1831,21 @@ def group_norm(context, node):
     input_shape = [*x.shape] # n, c, *
     num_groups = builtins.min(num_groups,c)
     new_shape = [n, num_groups, c//num_groups]
-    new_shape += [*x.shape[2:]] # adds remaining dims
+    if not any_symbolic(x.shape[2:]):
```

Comment:
Thanks for the fix @sercand,
it is very impressive that you are able to construct a complicated MI program from scratch :) 
I will suggest a more concise way to do:
Basically, our goal is to construct a `new_shape` Var, by doing the following operation:
```
new_shape = [n, num_groups, c//num_groups] + x.shape[2:]
```
This logic applies for both cases for symbolic or static input shape

Here is a psuedo code which is much more simpleand can be applied for both cases:
```
x_shape = mb.shape(x=x)
x_shape_sliced = mb.slice_by_size(x=x_shape, begin=[2], size=[-1]) # x_shape[2:]
new_shape = mb.concat(values=[[n, num_groups, c//num_groups], x_shape_slices], axis=0)
```
The above logic should produce exactly what we want :)

Reply:
@jakesabathia2 Thanks for the suggestion, it is much cleaner now. 

Also, I kept checking if there is any symbolic value in the shape in order to reduce the number of ops at output ml program. If you say that this optimization is not required and adding 3 extra ops is not a big deal, I can remove it.

Reply:
Yes, this is a nice check and please do keep it, thanks!

Reply:
Hi, I would like to ask: why do we need to keep the branch for 'if not any_symbolic' case ? It seemed that mb.shape() can cover both the dynamic and static shape cases.
Why not just wrap any input shape in mb.shape() ?

I'm mainly referring to this commit and trying to fix another bug, and am looking forward to some insights and suggestions, thanks.

Reply:
@xorange I commented the reason on the code: "optimization for non symbolic shapes. This get rids of 3 mil ops that required on dynamic shapes". It works but not using is faster, if it is not a dynamic shape.